common/tools/dev-tool/src/templates/sampleReadme.md.ts

```diff
@@ -241,7 +244,21 @@ Take a look at our [API Documentation][apiref] for more information about the AP
 ${info.customSnippets?.footer ?? ""}
 
 ${fileLinks(info)}
-[apiref]: ${info.apiRefLink ?? `https://learn.microsoft.com/javascript/api/@azure/${info.baseName}`}
+${await (async () => {
+  if (info.apiRefLink) {
+    return `[apiref]: ${info.apiRefLink}`;
+  } else if (info.scope.startsWith("@azure")) {
+    const link = `https://learn.microsoft.com/javascript/api/${info.scope}/${info.baseName}${info.isBeta ? "?view=azure-node-preview" : ""}`;
+    const res = await fetch(link);
+    if (res.ok) {
+      return `[apiref]: ${link}`;
+    }
```

Comment:
nitpick: could this be abstracted into a `isExistingUrl`? Could be helpful to check other URLs in the readme.

Reply:
extract into functions